src/drivers/uavcan/uavcan_drivers/socketcan/driver/src/socketcan.cpp

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ * NuttX SocketCAN port Copyright (C) 2022 NXP Semiconductors
+ *
+ *
+ *
+ */
+
+/****************************************************************************
+ *
+ *   Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ *   NuttX SocketCAN port Copyright (C) 2022 NXP Semiconductors
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <uavcan_nuttx/socketcan.hpp>
+#include <uavcan_nuttx/clock.hpp>
+#include <uavcan/util/templates.hpp>
+
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <string.h>
+
+#include <nuttx/can.h>
+#include <netpacket/can.h>
+
+#define MODULE_NAME "UAVCAN_SOCKETCAN"
+
+#include <px4_platform_common/log.h>
+
+namespace uavcan_socketcan
+{
+namespace
+{
+
+struct BitTimingSettings {
+	std::uint32_t canclkdiv;
+	std::uint32_t canbtr;
+
+	bool isValid() const { return canbtr != 0; }
+};
+
+} // namespace
+
+uavcan::uint32_t CanIface::socketInit(const char *can_iface_name)
+{
+
+	struct sockaddr_can addr;
+	struct ifreq ifr;
+
+	//FIXME Change this when we update to DroneCAN with CAN FD support
+	bool can_fd = 0;
```

Comment:
I think this answers the question I was about to ask - DroneCAN doesn't support CAN FD frames currently?  I know that Cyphal does...

Reply:
Master on https://github.com/dronecan/libuavcan supports CAN FD.
But the PX4 uavcan submodule hasn't been updated to support this.

Reply:
All you need is two boards of any type that support FD, and make one of them a node and the other the FMU.  Any of the STM32H7 variants support FD (though I still haven't found time to wrap up the SocketCAN support there for PX4).